### DISCUSSION TITLE
Expose createRemixRequest for testing

### DISCUSSION THREAD

**Discussion by andresgutgon** [remix-run/remix#2372] [General]
# Expose createRemixRequest
Hi, I'm working on some node code that interacts with Remix request. Maybe is not the right approach but I managed to do tests with [supertest](https://www.npmjs.com/package/supertest) + [express](https://www.npmjs.com/package/express) + `createRemixRequest`

So the idea is to be able to test `myThing` which expects a Remix request like this:
``` javascript
const action = async ({ request }) => {
  const result = myThing(request)
  // ... the rest of the code and send "result"
}
```

`myThing` has to be fully compliant with a remix request. So I did this in my tests. I created a Express server where I use my thing:
``` javascript
// ./testsHelpers/server
import express from "express"
import supertest from "supertest"

import { fixtures } from "../fixtures/form-data"
import { configParser } from "../testsHelpers/utils"

// This is code copied from remix-express:
// https://github.com/remix-run/remix/blob/main/packages/remix-express/server.ts#L96
// Because this function is not exported
import { createRemixRequest } from "./createRemixRequest"

const app = express()
app.get("/no-body-because-get", async (req, response) => {
  const request = createRemixRequest(req)
  const result await myThing.parse({
    request: request as unknown as Request
  })
  // ... the rest of a normal Express response to be check in the spec.
})

export const server = supertest(app)
```

The spec looks like this:
``` javascript
import { server } from "../testsHelpers/server"

describe("myThing", () => {
  describe("something that should happen", () => {
    test("throw error when content is undefined", async () => {
      const response = await server.get("/no-body-because-get")
      expect(response.status).toEqual(400)
      expect(response.body).toEqual({
        error: "Invariant failed: MIME type not provided in request"
      })
    })
  })
})
```

## The problem
One issue with this approach is that `createRemixRequest` [it's not exported](https://github.com/remix-run/remix/blob/main/packages/remix-express/index.ts). 

I was wondering 2 things here:
1. It's a good approach for testing this?
2. Would makes sense to export `createRemixRequest` to use it in cases like this?

Thanks!

**Comment by andresgutgon** [comment]
What I like about this approach is that it's a test agains a real remix request + express server.

![image](https://user-images.githubusercontent.com/49499/159002006-a49b2998-7200-427f-9ae4-1a3dd794d670.png)

Feels really nice

**Comment by MichaelDeBoey** [comment]
Going to close this in favor of #2481
